plugins/modules/lambda_layer_info.py

```diff
@@ -0,0 +1,189 @@
+        type: str
+        sample: "GPL-3.0-only"
+    compatible_architectures:
+        description: A list of compatible instruction set architectures.
+        returned: if it was defined for the layer version.
+        type: list
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+
+def list_layer_versions(module, lambda_client):
+
+    params = dict(LayerName=module.params.get("name"))
+    if module.params.get("compatible_runtime"):
+        params["CompatibleRuntime"] = module.params.get("compatible_runtime")
+    if module.params.get("compatible_architecture"):
+        params["CompatibleArchitecture"] = module.params.get("compatible_architecture")
+    try:
+        paginator = lambda_client.get_paginator('list_layer_versions')
+        layer_versions = paginator.paginate(**params).build_full_result()['LayerVersions']
+        return [camel_dict_to_snake_dict(layer) for layer in layer_versions]
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Unable to list layer versions for name {0}".format(module.params.get("name")))
+
+
+def list_layers(module, lambda_client):
+
+    params = dict()
+    if module.params.get("compatible_runtime"):
+        params["CompatibleRuntime"] = module.params.get("compatible_runtime")
+    if module.params.get("compatible_architecture"):
+        params["CompatibleArchitecture"] = module.params.get("compatible_architecture")
+    try:
+        paginator = lambda_client.get_paginator('list_layers')
+        layers = paginator.paginate(**params).build_full_result()['Layers']
+        layer_versions = []
+        for layer in layers:
+            layer.update(layer.pop("LatestMatchingVersion", None))
+            layer_versions.append(camel_dict_to_snake_dict(layer))
+        return layer_versions
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Unable to list layers {0}".format(params))
+
+
+def execute_module(module, lambda_client):
+
+    operation = list_layers
+    if module.params.get("name") is not None:
+        operation = list_layer_versions
+
+    module.exit_json(changed=False, layers_versions=operation(module, lambda_client))
+
+
+def main():
+    argument_spec = dict(
+        name=dict(type="str", aliases=["layer_name"]),
+        compatible_runtime=dict(type="str"),
+        compatible_architecture=dict(type="str"),
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    lambda_client = module.client('lambda', retry_decorator=AWSRetry.jittered_backoff())
```

Comment:
The retry decorator don't support paginators, we'll we need this to be separately handled.